app/src/main/kotlin/com/ayatk/biblio/repository/ranking/RankingRemoteDataSource.kt

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2016-2018 ayatk.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ayatk.biblio.repository.ranking
+
+import com.ayatk.biblio.data.narou.NarouClient
+import com.ayatk.biblio.data.narou.entity.NarouRanking
+import com.ayatk.biblio.data.narou.entity.enums.RankingType
+import com.ayatk.biblio.data.narou.util.QueryBuilder
+import com.ayatk.biblio.model.Novel
+import com.ayatk.biblio.model.Ranking
+import com.ayatk.biblio.model.enums.Publisher
+import io.reactivex.Single
+import java.util.Calendar
+import java.util.Date
+import javax.inject.Inject
+
+class RankingRemoteDataSource @Inject constructor(
+    private val narouClient: NarouClient
+) : RankingDataSource {
+
+  override fun getDailyRank(publisher: Publisher, range: IntRange): Single<List<Ranking>> {
+    val today = Calendar.getInstance()
+    // 午前6時以前にその日のランキングを取得するとエラーで死ぬので前日のランキングを取得
+    if (today.get(Calendar.HOUR_OF_DAY) < 6) {
+      today.add(Calendar.DATE, -1)
+    }
+
+    return narouClient.getRanking(today.time, RankingType.DAILY)
+        .flatMap {
+          val codes = it.map { it.ncode }.drop(range.first).take(range.count())
+          narouClient.getNovel(QueryBuilder().ncode(*codes.toTypedArray()).size(range.count()).build())
+              .map { novel -> convertRanking(it.drop(range.first).take(range.count()), novel) }
+        }
+  }
+
+  override fun getWeeklyRank(publisher: Publisher, range: IntRange): Single<List<Ranking>> {
+    val today = Calendar.getInstance()
+    // 午前6時以前にその日のランキングを取得するとエラーで死ぬので前週のランキングを取得
+    if (today.get(Calendar.HOUR_OF_DAY) < 6) {
```

Comment:
**[detekt]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.